### PR TITLE
feature(frontend): adding base logic for the data store config form

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the base logic for the data store config form.

## Changes

1. Plugin logic for the different data store types
2. Config draft object and reducer
3. Adding parsing services for the draft and initial values

## Fixes

- [[In-App Config] Setup Wizard#1622](https://github.com/kubeshop/tracetest/issues/1622)

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/src/components/DataStoreForm/DataStoreForm.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import {Form} from 'antd';
+import {useMemo} from 'react';
+import SetupConfigService from 'services/SetupConfig.service';
+import {TConfig, TDraftConfig, TDraftConfigForm} from 'types/Config.types';
+import DataStoreComponentFactory from '../DataStorePlugin/DataStoreComponentFactory';
+import * as S from './DataStoreForm.styled';
+
+export const FORM_ID = 'edit-test';
```
Comment (jorgeepc) [inline web/src/components/DataStoreForm/DataStoreForm.tsx:8]: let's change this `edit-test`

File: web/src/providers/SetupConfig/SetupConfig.provider.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  onReset: noop,
+});
+
+interface IProps {
+  children: React.ReactNode;
+}
+
+export const useSetupConfig = () => useContext(Context);
+
+const SetupConfigProvider = ({children}: IProps) => {
```
Comment (jorgeepc) [inline web/src/providers/SetupConfig/SetupConfig.provider.tsx:33]: What is the reason for having a new Provider? Could this be part of the `Config.provider`?

**Review by jorgeepc — APPROVED** [review]
Great job @xoscar. Added a couple of questions. Also, what do you think about moving all the config related components to `web/src/components/Settings/`? Not a big deal but we can analyze a new organization for the new features.

File: web/src/providers/SetupConfig/SetupConfig.provider.tsx (hunk 2 above)
Comment (xoscar) [reply web/src/providers/SetupConfig/SetupConfig.provider.tsx:33]: I was thinking that the ConfigProvider would be for read-only, while the setup config one is the one we'll be using to manipulate the config draft for each of the tabs and sections of the settings page. Does it make sense?

**Pull request merged**